src/Data/Binary/Builder/Base.hs

```diff
@@ -198,7 +198,7 @@ flush = Builder $ \ k buf@(Buffer p o u l) ->
       then k buf
       else let !b  = Buffer p (o+u) 0 l
                !bs = S.PS p o u
-           in return $! L.Chunk bs (inlinePerformIO (k b))
+           in return $! L.Chunk bs (accursedUnutterablePerformIO (k b))
```

Comment:
So the safety argument here is?

So it's ok to repeat these side effects right (ie in concurrent evaluation of the thunk) ? I think so because the place in the buffer where we write to is determined by the `b` value, and the writes should otherwise be deterministic.

I don't think it can be floated out and shared since the buffer references the incoming foreign ptr.

So yes i think this is ok.